infor-design/enterprise
# 3588/3615/3616 - Modal Q/A Fixes [v4.27.x]

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR fixes some regressions that appear to be the result of the Modal Buttonset changes and other Modal adjustments as of v4.27.x:
- Some Modal examples were not closing as expected when clicking their close buttons, or "Cancel" buttons inside the Modal Buttonset. This was due to a change made in Modals overall, because previously, clicking almost any buttons within the Modal Buttonset that had no callback functions associated with them would cause the Modal to incorrectly close.
- Some Modal examples configured not to destroy themselves would close, but unexpectedly not reopen on iOS devices.

**Related github/jira issue (required)**:
- Closes #3588
- Closes #3615
- Closes #3616

**Steps necessary to review your pull request (required)**:
Pull this branch, build, and run the demoapp.  Then:

_To test #3588:_
- Open http://localhost:4000/components/modal/example-close-btn.html in any browser
- Click the "Show Modal" button.  The Modal should open.
- Hover the "X" button.  It has a tooltip that should display.
- Click the "X" button.  When the modal closes, the tooltip should also immediately close.
- Re-open the Modal, and try to close it again by clicking the "X" button right before the tooltip appears.  The tooltip should never be visible when the modal is in the middle of closing, or is closed.

_To test #3615:_
- Open http://localhost:4000/components/modal/example-cancel-close.html in any browser
- Click the "Show Modal" button. The Modal should open.
- Clicking either button in the buttonset should cause a toast to fire, notifying the user that closing the Modal has been vetoed.
- Check the box above the buttonset labelled "Allow this Modal to close"
- Clicking either button in the buttonset should now close the Modal.

_To test #3616:_
- Open http://localhost:4000/components/modal/example-close-btn.html on an iOS device/simulator/browserstack.
- Click the "Show Modal" button. The Modal should open.
- Click either the "X" button, or either of the Modal Buttonset buttons.  The Modal should close.
- Click the "Show Modal" button again. The Modal should open again as expected.

**Included in this Pull Request**:
